package/kedro_viz/models/flowchart.py

```diff
@@ -337,10 +368,25 @@ class ModularPipelineNode(GraphNode):
     # when using the pipeline() factory function.
     # More information can be found here:
     # https://kedro.readthedocs.io/en/latest/06_nodes_and_pipelines/03_modular_pipelines.html#how-to-connect-existing-pipelines
-    internal_inputs: Set[str] = field(default_factory=set)
-    internal_outputs: Set[str] = field(default_factory=set)
-    external_inputs: Set[str] = field(default_factory=set)
-    external_outputs: Set[str] = field(default_factory=set)
+    internal_inputs: Set[str] = Field(
+        set(), description="The dataset inputs within the modular pipeline node"
+    )
+    internal_outputs: Set[str] = Field(
+        set(), description="The dataset outputs within the modular pipeline node"
+    )
+    external_inputs: Set[str] = Field(
+        set(),
+        description="""The dataset inputs connecting the modular
+        pipeline node with other modular pipelines""",
+    )
+    external_outputs: Set[str] = Field(
+        set(),
+        description="""The dataset outputs connecting the modular
+        pipeline node with other modular pipelines""",
+    )
+
+    # The type for Modular Pipeline Node
+    type: str = GraphNodeType.MODULAR_PIPELINE.value
```

Comment:
Why is this field shifted down?

Reply:
no big reason for the shift, but following the order from other classes. In pydantic validations happen in the order we define fields. Though we are not setting any value through pydantic validator for this field, I thought of having the enum types at the end.